### ISSUE TITLE
Azure - Migrate from EA Price Sheet API

### ISSUE THREAD

**Issue by raghavendra-rafay** [opencost/opencost#2826]
**Describe the bug**
[Azure has deprecated](https://learn.microsoft.com/en-us/azure/cost-management-billing/automate/migrate-ea-price-sheet-api) [Price Sheet API of Consumption Service](https://learn.microsoft.com/en-us/rest/api/consumption/price-sheet/download-by-billing-account-period?view=rest-consumption-2023-05-01&tabs=HTTP) which is being [used by Opencost](https://github.com/opencost/opencost/blob/develop/pkg/cloud/azure/pricesheetclient.go#L103) to fetch negotiated price.

**To Reproduce**
1. In an Enterprise Agreement Azure Account create an EKS Cluster with OpenCost.
2. Provide OfferId and Billing Account ID to OpenCost.
3. Check node_total_hourly_cost published by OpenCost.
4. Download the Cost and Usage Report from Azure's Cost Management.
5. Compare it with UnitPrice for the instances created by the EKS Cluster.

**Expected behavior**
node_total_hourly_cost should be equal to UnitPrice in Cost and Usage Report.

**Screenshots**
![Screenshot 2024-07-04 at 1 16 45 AM](https://github.com/opencost/opencost/assets/95902353/3adb241e-ca66-470c-b313-97ad1e3113b4)
![Screenshot 2024-07-04 at 12 50 00 AM](https://github.com/opencost/opencost/assets/95902353/7cc53fae-655f-490e-87d4-0895c2dda76f)
For the same meter-id UnitPrice is different.

**Which version of OpenCost are you using?**
v1.110

**Comment by mattray** [comment]
Is this redundant with https://github.com/opencost/opencost/issues/2247? It needs to be addressed either way.

**Comment by raghavendra-rafay** [comment]
@mattray No, #2247 suggests using the Consumption Price Sheet API. While Azure is suggesting to [migrate to Cost Management's Price Sheet API.](https://learn.microsoft.com/en-us/azure/cost-management-billing/automate/migrate-ea-price-sheet-api) 

We have started seeing differences in actual discounted prices and the Consumption Price Sheet API from May'24.

**Comment by enriqueavindi** [comment]
Hi! Is anyone still working on this issue? We need it in our organization and would be happy to help with it.

**Comment by enriqueavindi** [comment]
Hi! Is anyone working on this issue? We need this in our organization

**Comment by raghavendra-rafay** [comment]
@enriqueavindi To fix this issue, one needs an Enterprise Agreement with Microsoft. Unfortunately, we don't have that and are unable to proceed. Microsoft has implemented the current solution, which isn't valid anymore.

**Comment by nachopv** [comment]
Hi @raghavendra-rafay 
We can give it a try with our accounts to validate it, we would appreciate some guidance on which  parts of the code we should as long as we have never checked opencost code

- Detect if EA agreement applies
- Invoke the new API and format output properly

Does that make sense to you guys?

**Comment by raghavendra-rafay** [comment]
Hey @nachopv 
This is the trigger function to fetch pricing using the EA Pricing sheet API. 
https://github.com/opencost/opencost/blob/develop/pkg/cloud/azure/pricesheetdownloader.go#L32

Currently, it triggers
`/providers/Microsoft.Billing/billingAccounts/%s/billingPeriods/%s/providers/Microsoft.Consumption/pricesheets/download` which should change to `/providers/Microsoft.Billing/billingAccounts/%s/billingPeriods/%s/providers/Microsoft.CostManagement/pricesheets/download` [here](https://github.com/opencost/opencost/blob/develop/pkg/cloud/azure/pricesheetclient.go#L103).

**Comment by AjayTripathy** [comment]
Seems like a reasonable analysis would take a PR here!

**Comment by rwaal** [comment]
Hi everyone. Does anyone know if migration to the `Microsoft.CostManagement` Price Sheet API is planned for OpenCost?

Microsoft retired the older versions of the `Microsoft.Consumption` Azure Price Sheet on June 1, 2026. Including API version `2022-06-01`, which OpenCost currently seems to use.

As far as I understand, this Price Sheet integration is what allows OpenCost to use our customer-specific Azure Enterprise Agreement (EA) pricing instead of standard Azure rates. Is there any work underway to migrate this integration to the supported API?
